snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedTransientSnapshot.java

```diff
@@ -119,7 +126,7 @@ private void persistInternal(final CompletableActorFuture<PersistedSnapshot> fut
 
     if (!isValid) {
       future.completeExceptionally(
-          new IllegalStateException("Snapshot is not valid. It may have been deleted."));
+          new SnapshotNotFoundException("Snapshot is not valid. It may have been deleted."));
```

Comment:
:x: This message is confusing. The exception says not found, but the message says not valid, and it may have been deleted? What is true now?

Reply:
How about "Snapshot is not valid. Either it is not taken or it is already deleted."

Reply:
Ignore my previous comment. It does not reflect the correct state.

Reply:
At this point, if the snapshot is not valid, most likely the transient snapshot is already deleted because a newer snapshot was received. This is what we should convey here.

Reply:
Hm Ok I think I was confused because of the term "valid", because I thought it is related to the checksum or something. But it seems valid is just when it was taken successfully. Invalid if there was an error in between. Maybe we can just reuse the takenFuture? Or get the exception from this future and pass it into the given future? What do you think?

Reply:
It is also invalid if it was deleted before persisting.

Reply:
Ok this wouldn't be covered by the other future then right?

Reply:
See this change https://github.com/camunda-cloud/zeebe/pull/7922/commits/4e829e00fbb972048cb9bdcd90a9e1dcc7094239
Is it better?

Reply:
Hm I think isDone will return true, if it was completed exceptionally :)

Reply:
But yes I like it more to distinguish to get better messages.

Reply:
> Hm I think isDone will return true, if it was completed exceptionally 

Yes. This change https://github.com/camunda-cloud/zeebe/commit/4e829e00fbb972048cb9bdcd90a9e1dcc7094239 will cover it, no?